toolbox/utils.py

```diff
@@ -464,6 +466,55 @@ def as_command_choices(*args: t.Any, **kwargs: t.Any) -> t.Sequence[hikari.Comma
     return _list_to_command_choices(choices)
 
 
+def validate_embed(embed: hikari.Embed) -> hikari.Embed:
+    """Validate an embed, checking the length of all fields.
+    Parameters
+    ----------
+    embed : hikari.Embed
+        The embed to validate.
+    Raises
+    ------
+    EmbedValidationError
+        If the embed is invalid.
+    Returns
+    -------
+    hikari.Embed
+        The embed that was validated.
+    """
+    if (length := embed.total_length()) > 6000:
+        raise EmbedValidationError(f"Embed total length must be less than 6000 characters, got {length}.")
+
+    if embed.title and (length := len(embed.title)) > 256:
+        raise EmbedValidationError(f"Embed title must be less than 256 characters, got {length}.")
+
+    if embed.description and (length := len(embed.description)) > 4096:
+        raise EmbedValidationError(f"Embed description must be less than 4096 characters, got {length}.")
+
+    if embed.footer and embed.footer.text and (length := len(embed.footer.text)) > 2048:
+        raise EmbedValidationError(f"Embed footer text must be less than 2048 characters, got {length}.")
+
+    if embed.author and embed.author.name and (length := len(embed.author.name)) > 256:
+        raise EmbedValidationError(f"Embed author name must be less than 256 characters, got {length}.")
+
+    if embed.fields:
+        if (field_count := len(embed.fields)) > 25:
+            raise EmbedValidationError(f"Embed must have less than 25 fields, got {field_count}.")
+
+        for field in embed.fields:
+            if (length := len(field.name)) > 256:
+                raise EmbedValidationError(f"Embed field name must be less than 256 characters, got {length}.")
+            if (length := len(field.value)) > 1024:
+                raise EmbedValidationError(f"Embed field value must be less than 1024 characters, got {length}.")
+
+    if embed.author and embed.author.name and (length := len(embed.author.name)) > 256:
+        raise EmbedValidationError(f"Embed author name must be less than 256 characters, got {length}.")
+
+    if embed.footer and embed.footer.text and (length := len(embed.footer.text)) > 2048:
+        raise EmbedValidationError(f"Embed footer text must be less than 2048 characters, got {length}.")
```

Comment:
These are duplicate validations.

Reply:
I dont think they are

Reply:
I can't see any differences between lines 493-4 and 512-3 (same with 496-7 and 509-10), unless I'm missing something?

Reply:
Nvm they are duplicate

Reply:
Yeah that was a my b moment forgetting GitHub doesn't show the entire file in reviews lmao.